### PR TITLE
fix protobuf, containerd  vulns, add validate-fips step into onebranch build rp template 

### PR DESCRIPTION
**Pull request by bryanro92**
### Which issue this PR addresses:

Fixes FIPS Validation: https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/13418287/
Protobuf: https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/13312955 
Containerd: https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/13741090

I've combined the protobuf/containerd issues because they needed to be fixed in order for the OneBranch builds to pass to confirm the FIPS validation step is working.

### What this PR does / why we need it:

Adds `make validate-fips` step to the build template used for OneBranch pipeline which will validate we are using FIPS crypto libs.

Added exclude section in go.mod for flagged modules, and updated them with `go mod tidy`

### Test plan for issue:
`make validate-fips` already run in CI so that portion is tested. This PR has been run in OneBranch builds and is passing CI validation and module/component scanning.

### Is there any documentation that needs to be updated for this PR?
none that I can think of?

### CODE REVIEW

File: .pipelines/onebranch/templates/template-buildrp-buildaro.yml [hunk 1]
```diff
@@ -12,3 +12,11 @@ steps:
       mkdir -p $(ob_outputDirectory) 
       cp aro $(ob_outputDirectory)/aro
     workingDirectory: $(Build.SourcesDirectory)
+- task: Bash@3
```
Comment (bryanro92) [inline .pipelines/onebranch/templates/template-buildrp-buildaro.yml:15]: @mikeandescavage and concerns over this task being inserted here given the refactor you're doing in https://github.com/Azure/ARO-RP/pull/1991?

**Review by bennerv — APPROVED** [review]
lgtm

File: .pipelines/onebranch/templates/template-buildrp-buildaro.yml (hunk 1 above)
Comment (mikeandescavage) [reply .pipelines/onebranch/templates/template-buildrp-buildaro.yml:15]: No concerns here.  Once this is merged to master, we'll merge it into our working branch and integrate.

**Review by s-amann — APPROVED** [review]

File: go.mod [hunk 2]
```diff
@@ -130,12 +130,36 @@ require (
...
 	github.com/coreos/bbolt v1.3.3
 	github.com/etcd-io/bbolt v1.3.3
 	github.com/etcd-io/bbolt v1.3.6
+	// exclude github.com/golang/protobuf < 1.3.2 https://nvd.nist.gov/vuln/detail/CVE-2021-3121
+	github.com/gogo/protobuf v1.0.0
+	github.com/gogo/protobuf v1.1.1
+	github.com/gogo/protobuf v1.2.0
+	github.com/gogo/protobuf v1.2.1
+	github.com/gogo/protobuf v1.3.0
+	github.com/gogo/protobuf v1.3.1
```
Comment (m1kola) [inline go.mod:162]: See [this comment](https://github.com/Azure/ARO-RP/pull/1992#discussion_r818533904) form another PR:

> There are more [tags](https://github.com/gogo/protobuf/tags): do we need to exclude rest of them? E.g. `v0.5`, `v0.4`, etc

**Pull request merged**